File: Project/Web-SPA/src/app/entities/flight.ts
import { Time } from '@angular/common';
import { Destination } from './destination';
import { ChangeOver } from './changeOver';
import { Seat } from './seat';
import { Address } from './address';

export class Flight {
    airlineId: number;
    flightNumber: string;
    takeOffDate: Date;
    landingDate: Date;
    tripTime: string;
    tripLength: number;
    changeOverLocations: Array<ChangeOver>;
    ticketPrice: number;
    from: Address;
    to: Address;
    takeOffTime: string;
    landingTime: string;
    seats: Array<Seat>;

    constructor(airlineId: number, takeOffDate: Date, landingDate: Date, tripTime: string,
                tripLength: number, changeOverLocations: Array<ChangeOver>,
                ticketPrice: number, flightNumber: string,
                from: Address, to: Address, takeofftime: string, landingtime: string,
                seats: Array<Seat>) {
            this.airlineId = airlineId;
            this.takeOffDate = takeOffDate;
            this.landingDate = landingDate;
            this.tripTime = tripTime;
            this.tripLength = tripLength;
            this.changeOverLocations = changeOverLocations;
            this.ticketPrice = ticketPrice;
            this.flightNumber = flightNumber;
            this.from = from;
            this.to = to;
            this.landingTime = landingtime;
            this.takeOffTime = takeofftime;
            this.seats = seats;
    }
}

File: Project/WebApi/WebApi/Models/Flight.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace WebApi.Models
{
    public class Flight
    {
        public int FlightId { get; set; }
        public int AirlineId { get; set; }
        public Airline Airline { get; set; }
        public string FlightNumber { get; set; }
        public System.DateTime TakeOffDate { get; set; }
        public System.DateTime LandingDate { get; set; }
        public string TripTime { get; set; }
        public float tripLength { get; set; }
        public string TakeOffTime { get; set; }
        public string LandingTime { get; set; }
        public ICollection<FlightAddress> ChangeOvers { get; set; }
        public CityStateAddress From { get; set; }
        public CityStateAddress To { get; set; }
        public ICollection<Seat> Seats { get; set; }
    }
}
